### ISSUE TITLE
Cell voltage measurement at group boundary influenced by balancing of adjacent cell

### ISSUE THREAD

**Issue by frankrichter** [LibreSolar/BMS48V#4]
The cell voltage measurements of cell 5 and 6 (also cell 10 and 11 with bq76940) are influenced by the balancing current of the adjacent cell in the other cell group.
The voltage deviation depends on the resistance between BMS and cell, in my case it was about 15-20mV. With this error the BMS can't achieve an accurate balancing. 
This happens because the clocks of the 2 or 3 cell groups are not synchronized.

From "bq769x0 Family Top 10 Design Considerations" (SLUA749A) chapter 4 p. 9:
> The timelines of the cell groups in the bq76930 and bq76940 are independent, so one group may be balancing while the next group is measuring. At the cell boundary between groups, the adjacent cell may measure a voltage from the balance current in any common path. Keeping the common path resistance low, using wide traces or returning the current as close to the cell as possible will reduce this effect (see Figure 8).

Working solution/workaround: Remove the connections between VC5/VC5B and VC10/VC10B on the board and connect them to the respective cell with separate wires. Therefore the cells connector should be changed to a 20 pin model. 
With 10s version it is possible to use one of the spare pins C11-C15 (see attached photos of my hacked board):
![bms-hack_top](https://user-images.githubusercontent.com/13228024/41469807-f6147d6e-70ae-11e8-89af-4f633bb8ed4a.jpg)
![bms-hack_bottom](https://user-images.githubusercontent.com/13228024/41469844-13277352-70af-11e8-8069-06a45b6468f1.jpg)

**Comment by SolarDude** [comment]
frankrichter Nice catch!

So does your workaround as pictured also prevent this issue on Cell 10 & 11?

**Comment by frankrichter** [comment]
No, if you use the 15 cell version, you also have to split the C10
connection. And you can't use any spare pins in this case either.

We are currently working on a pull request which adresses this issue by
replacing the cells connector with a 20 pin model.

Am So., 20. Jan. 2019, 05:12 hat SolarDude <notifications@github.com>
geschrieben:

> frankrichter Nice catch!
>
> So does your workaround as pictured also prevent this issue on Cell 10 &
> 11?

**Comment by SolarDude** [comment]
I want to make the changes on my board so the cell voltage is as accurate as possible. 

The BQ76940 EVM is only using 16 cell input pins so are you saying that this issue voltage drop during balancing issue may be happening on the EVM module also? Or maybe not as noticeable because the EVM board is not setup with resistors to balance at 150mA which would cause a less of a voltage drop? 

![image](https://user-images.githubusercontent.com/7421684/51443602-29f59a80-1cb9-11e9-9be7-eb386b5d3378.png)

I think maybe the smaller traces running to the balancing resistors & circuit could also be adding to the voltage drop during balancing. You can see how wide the traces are that run to the balancing resistors on their EVM boards. 

![image](https://user-images.githubusercontent.com/7421684/51444004-b1460c80-1cbf-11e9-844c-fbd8ea5c310d.png)

It seems like if we want to keep the board small and not increase the width of the traces running from the 18 or a new 20 pin connector to the balancing resistors then running a extra voltage sense wire to VC5B & VC10B pins would provide the most accurate cell voltage readings and balancing performance. 

Took me awhile to wrap my head around what your workaround but now I understand what is going on. 

I'm going to make these changes today. I've imported the Eagle project to KiCad 5.0.2-1 and that went well except for most of the Global Labels were just text so I have to change all those to proper Global Labels and the part text indicators needed moved around so they were not directly under some parts. 

![image](https://user-images.githubusercontent.com/7421684/51444259-75ad4180-1cc3-11e9-8bdd-8d6b1ca7ba0c.png)

I'm adding the bq78350 chip to the mix the same as the EVM modules and I'm using the Particle Photon Wifi MCU since I'm used to using that module. 

Do you see any reason this 48v board would not be usable with either the BQ76920, BQ76930, or BQ76940 AFE chips? It would be nice for one board to be able to handle 4, 8, 15 cell LiFePo4 battery Packs.

**Comment by martinjaeger** [comment]
Very nice that you are transferring the board to KiCad. We could open a separate branch of the Particle Photon version or open a new repository, if you like.

Unfortunately, the bq76920 has a different pin pitch than the other two, so this this board currently supports only the 30 and 40 chips. I'm using it with the bq76930 and 8 cells.
The [TI top 10 design considerations document](http://www.ti.com/lit/an/slua749a/slua749a.pdf) makes a suggestion for board layout to support all 3 chip types. But it doesn't allow any traces below the chip anymore.

**Comment by frankrichter** [comment]
> It seems like if we want to keep the board small and not increase the width of the traces running from the 18 or a new 20 pin connector to the balancing resistors then running a extra voltage sense wire to VC5B & VC10B pins would provide the most accurate cell voltage readings and balancing performance.

Yes, even with wider traces the measurements will still be affected by the voltage drop caused by the connector and the wire between BMS and cell.
Separate sense wires will do the trick regardless of the resistance and the applied balancing current. That's why I went for this solution.

**Comment by SolarDude** [comment]
> 
> 
> > It seems like if we want to keep the board small and not increase the width of the traces running from the 18 or a new 20 pin connector to the balancing resistors then running a extra voltage sense wire to VC5B & VC10B pins would provide the most accurate cell voltage readings and balancing performance.
> 
> Yes, even with wider traces the measurements will still be affected by the voltage drop caused by the connector and the wire between BMS and cell.
> Separate sense wires will do the trick regardless of the resistance and the applied balancing current. That's why I went for this solution.

Makes sense. 

The 20 pin connector can be added pretty easily without needing to move the current temp sensor port. 

![image](https://user-images.githubusercontent.com/7421684/51445758-b498c280-1cd6-11e9-9dfd-0c170be4515f.png)

![image](https://user-images.githubusercontent.com/7421684/51445832-b1520680-1cd7-11e9-8601-128964136763.png)


The new last 2 pins on the left of cell port just need to be connected to VC5B & VC10B to fix the issue. 

I need to change all these global labels on the KiCad conversion before I can properly draw new wires to the new pins but the change is really easy. I'll report back once I get this done.

**Comment by frankrichter** [comment]
> The new last 2 pins on the left of cell port just need to be connected to VC5B & VC10B to fix the issue.

Don't connect the new pins to VC5B and VC10B directly, this won't give the desired effect.

You have to remove the connection between R40 and R41 and connect the left side of R41 with an additional wire to cell 5 (positive). Same for R46/R47.

**Comment by SolarDude** [comment]
> 
> 
> Very nice that you are transferring the board to KiCad. We could open a separate branch of the Particle Photon version or open a new repository, if you like.
> 
> Unfortunately, the bq76920 has a different pin pitch than the other two, so this this board currently supports only the 30 and 40 chips. I'm using it with the bq76930 and 8 cells.
> The [TI top 10 design considerations document](http://www.ti.com/lit/an/slua749a/slua749a.pdf) makes a suggestion for board layout to support all 3 chip types. But it doesn't allow any traces below the chip anymore.

I just read that document some more about the differences between the chips and it does look like it's not a good idea to try to make the board work with all 3 AFE chips due to the differences between the 4 cell and higher cell count designs.

**Comment by SolarDude** [comment]
> 
> 
> > The new last 2 pins on the left of cell port just need to be connected to VC5B & VC10B to fix the issue.
> 
> Don't connect the new pins to VC5B and VC10B directly, this won't give the desired effect.
> 
> You have to remove the connection between R40 and R41 and connect the left side of R41 with an additional wire to cell 5 (positive). Same for R46/R47.

Your talking about cutting where you see the Blue X marks and adding where you see the bright RED lines? 

Then connect the new C5+ & C10+ connections to where you see the White C5+ and C10+ text right? 

![image](https://user-images.githubusercontent.com/7421684/51446249-705cf080-1cdd-11e9-90eb-dc594c8eb064.png)

**Comment by frankrichter** [comment]
Your cuts are ok, your new connections aren't. You have to connect C5/C10 to the lower pad of R40/46, not to the upper pad.
Please refer to the photo in my first post. There is no connection between the enameled copper wire and the trace below.

**Comment by SolarDude** [comment]
> 
> 
> Your cuts are ok, your new connections aren't. You have to connect C5/C10 to the lower pad of R40/46, not to the upper pad.
> Please refer to the photo in my first post. There is no connection between the enameled copper wire and the trace below.

Got it!

We need the cell to feed through the resistors before going into the AFE pins. 
 
So we need the new connections that are shown by the new dark blue lines going from C5 & C10 right?

![image](https://user-images.githubusercontent.com/7421684/51446508-1e1dce80-1ce1-11e9-8f20-31c4cba52482.png)

**Comment by frankrichter** [comment]
Yes, this is how we modded our board. Since then it is working flawlessly.

Additionnally it would make sense to connect the C5 and C10 vias to C5/C10 and not to the newly created C5+/C10+ like in your image, otherwise it won't work if you connect only 10 cells and leave out the C10+ connection which is not necessary for 10 cells.

The schematic should look like this:
[LibreSolar_BMS48V_20190121.pdf](https://github.com/LibreSolar/BMS48V/files/2777161/LibreSolar_BMS48V_20190121.pdf)

**Comment by SolarDude** [comment]
> 
> 
> Yes, this is how we modded our board. Since then it is working flawlessly.
> 
> Additionnally it would make sense to connect the C5 and C10 vias to C5/C10 and not to the newly created C5+/C10+ like in your image, otherwise it won't work if you connect only 10 cells and leave out the C10+ connection which is not necessary for 10 cells.

Great, makes sense to connect to the Vias as you did in your mod.

I think I know exactly what is going on now and can properly make the changes. 

What is your battery pack setup? How much current? Which Sense board are you using?

**Comment by SolarDude** [comment]
Looks like adding the footprint for the BQ76920 would not require to much modification based on what I'm seeing here: 

![image](https://user-images.githubusercontent.com/7421684/51446750-4e1aa100-1ce4-11e9-9f82-d8651fd13d60.png)

![image](https://user-images.githubusercontent.com/7421684/51446755-5e328080-1ce4-11e9-91b7-c37bae53cedf.png)

I'm going attempt to add the footprint and make this a board that can handle the whole range of normal battery pack configurations.

**Comment by frankrichter** [comment]
> Great, makes sense to connect to the Vias as you did in your mod.

In the mod the via is connected to C5+/C10+ like in your image. In a new layout this should be avoided.

> What is your battery pack setup?

9 of these cells: https://www.lithiumakkushop.de/product/innopower-lfmp-60ah/

> How much current?

about 30A

> Which Sense board are you using?

https://github.com/LibreSolar/Switch-N-Sense

**Comment by SolarDude** [comment]
> 
> 
> Yes, this is how we modded our board. Since then it is working flawlessly.
> 
> Additionnally it would make sense to connect the C5 and C10 vias to C5/C10 and not to the newly created C5+/C10+ like in your image, otherwise it won't work if you connect only 10 cells and leave out the C10+ connection which is not necessary for 10 cells.
> 

So just so we are clear. 

You are saying the new C5+ and C10+ should connect at the vias that the orange arrows are pointing to? 

![image](https://user-images.githubusercontent.com/7421684/51446932-81f6c600-1ce6-11e9-82a9-0106633c238c.png)

**Comment by SolarDude** [comment]
> 
> 
> > Yes, this is how we modded our board. Since then it is working flawlessly.
> > Additionnally it would make sense to connect the C5 and C10 vias to C5/C10 and not to the newly created C5+/C10+ like in your image, otherwise it won't work if you connect only 10 cells and leave out the C10+ connection which is not necessary for 10 cells.
> 
> So just so we are clear.
> 
> You are saying the new C5+ and C10+ should connect at the vias that the orange arrows are pointing to? If not what is recommended? 
> 
> ![image](https://user-images.githubusercontent.com/7421684/51446932-81f6c600-1ce6-11e9-82a9-0106633c238c.png)

**Comment by SolarDude** [comment]
> 
> 
> > Great, makes sense to connect to the Vias as you did in your mod.
> 
> In the mod the via is connected to C5+/C10+ like in your image. In a new layout this should be avoided.
> 
> > What is your battery pack setup?
> 
> 9 of these cells: https://www.lithiumakkushop.de/product/innopower-lfmp-60ah/
> 
> > How much current?
> 
> about 30A
> 
> > Which Sense board are you using?
> 
> https://github.com/LibreSolar/Switch-N-Sense

What kind of heat does the sense board produce at 30 amps? 

How much current testing have you done with the board? Just curious what is possible. 

I have one on order with 2oz copper on both sides of the board.

**Comment by frankrichter** [comment]
> You are saying the new C5+ and C10+ should connect at the vias that the orange arrows are pointing to?

Nope, this is how it should not be done...

> If not what is recommended?

Connect them to the right, so they can stay on the respective C5/C10 rail.

**Comment by frankrichter** [comment]
> What kind of heat does the sense board produce at 30 amps?

No measurements, but no problems so far.

> How much current testing have you done with the board?

No current testing done so far, but the board is rated with 80A. Perhaps you need a heatsink at full load.
But be carefull when not all MOSFETs are switched on, the voltage drop at the body diode will produce a lot more heat.

It's late now in Central Europe, let's talk tomorrow.

**Comment by SolarDude** [comment]
> > If not what is recommended?
> 
> Connect them to the right, so they can stay on the respective C5/C10 rail.

I'm not sure why but I'm totally confused

Can you post a picture showing where the new C5+ and C10+ lines should connect to on the PCB board picture I posted?

I want to make sure it's right.

**Comment by frankrichter** [comment]
I will do a quick drawing in the evening, but it's not as complicated as you think ;-)

**Comment by SolarDude** [comment]
I think I got the new 20 Pin connector added and the wiring modified to prevent the cell voltage offset issue from happening. Take a look at the schematic and let me know if you see anything wrong. 

I made sure I copied the new schematic you posted so we are on the same page as far as the C5 & C10 connections. 

![image](https://user-images.githubusercontent.com/7421684/51488995-04c66200-1d75-11e9-84d7-e336c31a38de.png)

![image](https://user-images.githubusercontent.com/7421684/51489004-0abc4300-1d75-11e9-9ec5-1dc4b26b608a.png)

![image](https://user-images.githubusercontent.com/7421684/51489013-10198d80-1d75-11e9-929d-d8df1a5888fd.png)

**Comment by frankrichter** [comment]
Looks good!

**Comment by photon-delight** [comment]
> Very nice that you are transferring the board to KiCad. We could open a separate branch of the Particle Photon version or open a new repository, if you like.

Just wondering whether the KiCad version ever made it into a repo?

**Comment by MichiS79** [comment]
I would also like to re-build the BMS and curious if there any updates. @SolarDude Could share your the KiCad version? 
Thx a lot.

**Comment by utshab01** [comment]
https://batterlution.com/bms-problems-solved-fast-from-balcony-to-industrial-energy-storage/#Common_BMS_Problems_and_Solutions_for_Balcony_battery_storage_Residential_Commercial_and_Industrial_Energy_Storage
